### DISCUSSION TITLE
Automatically create a note for a YouTube link on your clipboard

### DISCUSSION THREAD

**Discussion by marcus-crane** [SilentVoid13/Templater#631] [Templates Showcase]
Hi there,

I've been meaning to check out Templater's deeper Javascript functionality and it's pretty handy! While I'm still integrating Obsidian into my daily workflow, one thing I've been doing is using Obsidian to take notes on YouTube videos.

I had previously used the Rich Links extension but I realised that I should make use of the embed support so I can actually take notes while watching the video inside the note itself.

https://user-images.githubusercontent.com/14816406/167294786-56105456-5ae1-41db-ba83-725186d6131b.mp4

The script is nothing fancy since I threw it together in a couple of minutes. It needs proper error handling but it's useful as a start anyway.

*Scripts/youtube.js*
https://gist.github.com/marcus-crane/f2ef3863336e1c53a0b4d1968a8715b5

*Templates/YouTube*
```html
<iframe src="https://www.youtube-nocookie.com/embed/<% tp.user.youtube(tp) %>?vq=hd1080&modestbranding=1&rel=0&iv_load_policy=3" width="569" height="317" frameborder="0"></iframe>
```

I'm sure it'll break when I run into some issue. I figure it might be a neat addition to the core web module if there's some interest? It would need to be made a bit more robust of course.

**Comment by AB1908** [comment]
This is cool, thanks! Any ideas for doing timestamps and stuff?

**Reply by marcus-crane** [reply]
Could you clarify a bit which timestamps you have in mind? 

ie; the video length, description and so on or something else?

**Reply by AB1908** [reply]
Yea like taking notes for a particular point in the video.

**Reply by marcus-crane** [reply]
Ah, that would be well outside of the scope of Templater and likely it's own plugin of anything but I don't have any plans to create anything like that.

#I can recommend https://chrome.google.com/webstore/detail/reclipped-take-notes-on-y/gbnebpdekafhpcipejfhabfghccgfnbh?hl=en which can indirectly sync to Obsidian via Readwise though

**Reply by AB1908** [reply]
Thanks for the info there!
